src/main/java/teammates/ui/controller/InstructorCourseDetailsPageData.java

```diff
@@ -20,9 +24,132 @@ public InstructorCourseDetailsPageData(AccountAttributes account) {
+            
+            ElementTag allRecordsButton = createButton("All Records", "btn btn-default btn-xs", null, this.getStudentRecordsLink(student),
+                                                   Const.Tooltips.COURSE_STUDENT_RECORDS, "tooltip", null, false);
+
+            row.actions.add(viewButton);
+            row.actions.add(editButton);
+            if (!student.isRegistered()) {
+                row.actions.add(sendInviteButton);
+            }
+            row.actions.add(deleteButton);
+            row.actions.add(allRecordsButton);
+            
+            isDisabled = !this.currentInstructor.isAllowedForPrivilege(student.section, Const.ParamsNames.INSTRUCTOR_PERMISSION_GIVE_COMMENT_IN_SECTIONS);
+            ElementTag addCommentButton = createButton("Add Comment", "btn btn-default btn-xs cursor-default", null, "javascript:;",
+                                                       Const.Tooltips.COURSE_STUDENT_COMMENT, "tooltip", null, isDisabled);
+            
+            isDisabled = !this.currentInstructor.isAllowedForPrivilege(student.section, Const.ParamsNames.INSTRUCTOR_PERMISSION_GIVE_COMMENT_IN_SECTIONS);
+            ElementTag addCommentsButton = createButton("<span class=\"caret\"></span><span class=\"sr-only\">Add comments</span>", 
+                                                        "btn btn-default btn-xs dropdown-toggle", null, "javascript:;",
+                                                        null, "dropdown", null, isDisabled);
+            row.commentActions.add(addCommentButton);
+            row.commentActions.add(addCommentsButton);
+            
+            ElementTag toStudentCommentOption = createButton("To student: " + sanitizeForHtml(row.student.name), 
+                                                             "t_student_details_tostudent-c" + this.courseDetails.course.id + "." + idx, null,
+                                                             getCourseStudentDetailsLink(row.student) + "&" + Const.ParamsNames.SHOW_COMMENT_BOX + "=student", 
+                                                             null, null, null, false);
+            toStudentCommentOption.setAttribute("role", "menuitem");
+            toStudentCommentOption.setAttribute("tabindex", "-1");
+            
+            ElementTag toTeamCommentOption = createButton("To team: " + sanitizeForHtml(row.student.team), 
+                                                          "t_student_details_toteam-c" + this.courseDetails.course.id + "." + idx, null,
+                                                          getCourseStudentDetailsLink(row.student) + "&" + Const.ParamsNames.SHOW_COMMENT_BOX + "=team", 
+                                                          null, null, null, false);
+            toTeamCommentOption.setAttribute("role", "menuitem");
+            toTeamCommentOption.setAttribute("tabindex", "-1");
```

Comment:
There is another way is putting those attributes to the .tag file, next to the forEach. I find it not worth of introducing more parameters in the createButton() function because it have already had so many attributes.